src/apps/backend/modules/todo/internal/todo_writer.py

```diff
@@ -0,0 +1,61 @@
+from dataclasses import asdict, replace
+from datetime import datetime
+
+from bson import ObjectId
+from pymongo import ReturnDocument
+
+from modules.todo.errors import InvalidDueDateFormatError, TodoNotFoundError
+from modules.todo.internal.store.todo_model import TodoModel
+from modules.todo.internal.store.todo_repository import TodoRepository
+from modules.todo.types import CreateTodoParams, Todo, UpdateTodoParams
+
+
+class TodoWriter:
+    @staticmethod
+    def create_todo(account_id: str, params: CreateTodoParams) -> Todo:
+        if isinstance(params.due_date, str):
+            try:
+                parsed_due_date = datetime.fromisoformat(params.due_date)
+            except ValueError:
+                raise InvalidDueDateFormatError(params.due_date)
+            params = replace(params, due_date=parsed_due_date)
+
+        todo_bson = TodoModel(
+            id=None,
+            title=params.title,
+            description=params.description,
+            status=params.status,
+            due_date=params.due_date,
+            account_id=account_id,
+        ).to_bson()
+
+        result = TodoRepository.collection().insert_one(todo_bson)
+        return Todo(
+            id=str(result.inserted_id),
+            title=params.title,
+            description=params.description,
+            status=params.status,
+            due_date=params.due_date,
+        )
+
+    @staticmethod
+    def update_todo(todo_id: str, params: UpdateTodoParams) -> Todo:
+        update_data = {k: v for k, v in asdict(params).items() if v is not None}
```

Comment:
Why are we filtering `None`? A caller may want to unset the value and that should be a valid operation